lib/types.nix

```diff
@@ -113,4 +113,32 @@ rec {
         types.optionDescriptionPhrase (class: class == "noun" || class == "composite") elemType
       }";
     };
+
+  pluginLuaConfig = types.submodule (
+    { config, ... }:
+    {
+      options = {
+        pre = lib.mkOption {
+          type = types.lines;
+          default = "";
+          description = "Configuration before the initialization of the plugin";
+        };
+        post = lib.mkOption {
+          type = types.lines;
+          default = "";
+          description = "Configuration after the initialization of the plugin";
+        };
+        init = lib.mkOption {
+          type = types.lines;
+          default = "";
+          description = "Initialization code (between pre & post)";
```

Comment:
Same;

"between" pre&post is actually a little misleading. Anyone that _reads_ `init` will see it as _including` pre&post.

Anyone who uses `mkOrder` on init may also be confused when pre&post get merged in-between their ordered definitions.

Reply:
Hmm... this makes me not like tying them to the same mkOrder merge if we can have this interaction. It should be set that pre comes before init and post comes after.

Reply:
Yeah I think I should change the name, to something like `config`  or `content`